inference-engine/thirdparty/clDNN/src/gpu/kernels_cache.cpp

```diff
@@ -28,13 +28,89 @@
 
 #include "kernel_selector_helper.h"
 
-#define MAX_KERNELS_PER_PROGRAM 10
-
-namespace cldnn {
-namespace gpu {
+#ifndef ENABLE_UNICODE_PATH_SUPPORT
+# ifdef _WIN32
+#  if defined __INTEL_COMPILER || defined _MSC_VER
+#   define ENABLE_UNICODE_PATH_SUPPORT
+#  endif
+# elif defined(__GNUC__) && (__GNUC__ > 5 || (__GNUC__ == 5 && __GNUC_MINOR__ > 2)) || defined(__clang__)
+#  define ENABLE_UNICODE_PATH_SUPPORT
+# endif
+#endif
+
+#ifndef _WIN32
+#ifdef ENABLE_UNICODE_PATH_SUPPORT
+#include <locale>
+#include <codecvt>
+#endif
+#else
+#include <Windows.h>
+#endif
 
 namespace {
-std::string get_undef_jit(kernels_cache::source_code org_source_code) {
+
+std::mutex cacheAccessMutex;
+
+#ifdef ENABLE_UNICODE_PATH_SUPPORT
+std::wstring multiByteCharToWString(const char* str) {
+#ifdef _WIN32
+    int strSize = static_cast<int>(std::strlen(str));
+    int size_needed = MultiByteToWideChar(CP_UTF8, 0, str, strSize, NULL, 0);
+    std::wstring wstrTo(size_needed, 0);
+    MultiByteToWideChar(CP_UTF8, 0, str, strSize, &wstrTo[0], size_needed);
+    return wstrTo;
+#else
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> wstring_encoder;
+    std::wstring result = wstring_encoder.from_bytes(str);
+    return result;
+#endif  // _WIN32
+}
+#endif  // ENABLE_UNICODE_PATH_SUPPORT
+
+static std::vector<unsigned char> loadBinaryFromFile(std::string path) {
+    std::lock_guard<std::mutex> lock(cacheAccessMutex);
+
+#if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    std::wstring widefilename = multiByteCharToWString(path.c_str());
+    const wchar_t* filename = widefilename.c_str();
+    FILE *fp = _wfopen(filename, L"rb");
+#else
+    const char* filename = path.c_str();
+    FILE *fp = fopen(filename, "rb");
+#endif
+
+    if (fp) {
+        fseek(fp, 0, SEEK_END);
+        size_t nsize = (size_t)ftell(fp);
+
+        fseek(fp, 0, SEEK_SET);
+
+        std::vector<unsigned char> ret(nsize);
+
+        auto res = fread(ret.data(), sizeof(unsigned char), nsize, fp);
+        (void)res;
+        fclose(fp);
+        return ret;
+    }
+
+    return {};
+}
+
+static void saveBinaryToFile(std::string path, const std::vector<unsigned char> buffer) {
```

Comment:
can we add a test for such case? See `smoke_registerPluginsXMLUnicodePath` for example..

Reply:
added test with unicode characters in path